### ISSUE TITLE
How to get features vector of detection (bbox, class, confidence) ?

### ISSUE THREAD

**Issue by folkien** [AlexeyAB/darknet#7291]
Hi,

Is there possible to get from darknet `features vector` for each YOLOv4 detection  `(bbox, class, confidence)` ? 
I've like to use it during video processing for my multiple object tracking algorithm.

**Comment by shubham-shahh** [comment]
you are planning to run YOLO on images or videos? and are you using the python or C code to run your model?

**Comment by folkien** [comment]
Video. At the moment i use python with wrapper for library file darknet.so

**Comment by shubham-shahh** [comment]
Do you want to save it to a .txt file or simply print it on the screen?

**Comment by folkien** [comment]
I need realtime acces to `features vector` for each detection for each video frame.

**Comment by shubham-shahh** [comment]
you can access it by accessing the elements of the detection queue

**Comment by folkien** [comment]
Could you give me example code how to get it?

**Comment by shubham-shahh** [comment]
when I am printing the first element of the detections queue i am getting 

`[('dog', '63.29', (280.6484069824219, 187.42095947265625, 80.493896484375, 47.97364044189453))]
`

the format is "obj name", "confidence", "co-ordinates"

is that what you are expecting?

**Comment by folkien** [comment]
No. This is not what i need, because I have this now. I need 'features vector' the 'vector' of Net values before classification. Layer of neural network before the last one layer.

**Comment by shubham-shahh** [comment]
okay gotcha, ill look into it and let you know and by the way if you are using yolo with openCV it is really simple. refer this video [https://www.youtube.com/watch?v=9AycYn9gj1U](url)

**Comment by folkien** [comment]
Thanks this video looks good. Unfortunately i'm using darknet.so and darknet python wrapper to handle YOLO model. So i need similar functions to get data from inside darknet.

**Comment by shubham-shahh** [comment]
But, that's what you need right like the output he extracted from 3 different points from the network?

**Comment by laplaceson** [comment]
Hi @folkien , Do you resolve ths issue, i also have same question. Thanks

**Comment by folkien** [comment]
@laplaceson @shubham-shahh No, finally i haven't tested https://www.youtube.com/watch?v=9AycYn9gj1U approach with open cv.

**Comment by laplaceson** [comment]
Thanks , But this is nothing help actually, it only show how to extract bbox but not its corresponding feature vector.

**Comment by xuaner11111** [comment]
hello everyone, do you solve this problem? get features vector?

**Comment by akashAD98** [comment]
im also looking for same , i want bbox coordinates without NMS ? is there any way to get it?

**Comment by folkien** [comment]
> im also looking for same , i want bbox coordinates without NMS ? is there any way to get it?

Actaully this is simple. You have to disable call 

```python
        do_nms_sort(detections, num, len(class_names), nms)
```

in method `detect_image` in darknet.py
